domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterModuleTest.kt

```diff
@@ -0,0 +1,86 @@
+package org.oppia.android.domain.platformparameter
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+import org.oppia.android.testing.platformparameter.FakePlatformParameterModule
+import org.oppia.android.testing.platformparameter.TEST_PARAM_DEFAULT_VALUE
+import org.oppia.android.testing.platformparameter.TEST_PARAM_NAME
+import org.oppia.android.testing.platformparameter.TEST_PARAM_VALUE
+import org.oppia.android.testing.platformparameter.TestParam
+import org.oppia.android.util.platformparameter.PlatformParameter
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterModuleTest.TestApplication::class)
+class PlatformParameterModuleTest {
+
+  @Inject
+  @TestParam
+  lateinit var testPlatformParameter: PlatformParameter<String>
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testFakeModule_initPlatformParameterMap_retrieveTestParameter_verifyItsValue() {
+    val testParam = ParameterValue.newBuilder().setString(TEST_PARAM_VALUE).build()
+    platformParameterSingleton.setPlatformParameterMap(
+      mapOf(
+        TEST_PARAM_NAME to testParam
+      )
+    )
+    assertThat(testPlatformParameter.value).isEqualTo(TEST_PARAM_VALUE)
+  }
```

Comment:
@jcqli I will need some help over this. 
Aim --> What I am trying to do here is that I want to initialise the platformParameterMap in the PlatformPrameterSingleton so that FakePlatformParameterModule can retrieve the value of TestParam from it and provide it via Dagger to us. 
Problem --> As the **setUpTestApplication**( ) is already called before every test (becuase it is in the @Before anotated method) hence the **testPlatformParameter** is injected before any test starts (which in turn injects the default parameter because the **platformParameterSingleton** initally have its **platformParameterMap** to be empty). This hinders the test to inject a platform parameter at runtime. 
Tried solution --> I have tried to call the **setUpTestApplication** later in the tests so that **platformParameterSingleton** can be initialised before the **testPlatformParameter** is injected , but this method fails as the platformParameterSingleton is itself injected.

Reply:
Hm I think what you want to do is actually use the real platform parameter module with a test singleton. Since this it the PlatformParameterModuleTest, we want to make sure that our production module is functioning as expected. That said, the PlatformParameterModule is currently empty so it's unclear what needs to be tested right now.

Reply:
We have implmented new Testing approach now which doesn't require us to have a Fake Singleton at all. Kindly take a look

Reply:
Nice, looks good!